src/perception/tag_detector/tag_detector.processing.cpp

```diff
@@ -84,33 +87,77 @@ namespace mrover {
 
         // Update ID, image center, and increment hit count for all detected tags
         for (size_t i = 0; i < mIds.size(); ++i) {
-          int id = mIds[i];
-          Tag& tag = mTags[id];
-          tag.hitCount = std::clamp(tag.hitCount + 1, 0, mMaxHitCount);
-          tag.id = id;
-          tag.imageCenter = std::accumulate(mCorners[i].begin(), mCorners[i].end(), cv::Point2f{}) / static_cast<float>(mCorners[i].size());
-          tag.tagInCam = getFidInCamFromPixel(msg, std::lround(tag.imageCenter.x), std::lround(tag.imageCenter.y));
-
-          if (tag.tagInCam) {
-            // Publish tag to immediate
-            std::string immediateFrameId = "immediateFiducial" + std::to_string(tag.id);
-            SE3::pushToTfTree(mTfBroadcaster, immediateFrameId, mCameraFrameId, tag.tagInCam.value());
-          }
+            int id = mIds[i];
+            Tag& tag = mTags[id];
+            tag.hitCount = std::clamp(tag.hitCount + 1, 0, mMaxHitCount);
+            tag.id = id;
+            tag.imageCenter = std::accumulate(mCorners[i].begin(), mCorners[i].end(), cv::Point2f{}) / static_cast<float>(mCorners[i].size());
+
+            // find extremes of detected tag for checking later
+            auto xBounds = std::minmax_element(mCorners[i].begin(), mCorners[i].end(),
+                                               [](const cv::Point2f& a, const cv::Point2f& b) -> bool {
+                                                   return a.x < b.x;
+                                               });
+
+            int leftmost = (*xBounds.first).x;
```

Comment:
Use https://en.cppreference.com/w/cpp/language/structured_binding to unpack the pair, and then operate on that

Reply:
e.g. `auto const& [minIt, maxIt] = xBounds`

Reply:
Then access with `minIt->x` for example